src/Shell/Task/ModelTask.php

```diff
@@ -85,6 +85,14 @@ public function main($name = null)
         parent::main();
         $name = $this->_getName($name);
 
+        if (!empty($this->connection)) {
+            $connection = ConnectionManager::get($this->connection);
+            $config = $connection->config();
+            if (isset($config['prefix'])) {
+                $this->tablePrefix = $config['prefix'];
+            }
+        }
```

Comment:
I would much rather provide the prefix via a command line option, instead of the connection config.